examples/bicluster/plot_spectral_biclustering.py

```diff
@@ -3,63 +3,111 @@
 A demo of the Spectral Biclustering algorithm
 =============================================
 
-This example demonstrates how to generate a checkerboard dataset and
-bicluster it using the Spectral Biclustering algorithm.
-
-The data is generated with the ``make_checkerboard`` function, then
-shuffled and passed to the Spectral Biclustering algorithm. The rows
-and columns of the shuffled matrix are rearranged to show the
-biclusters found by the algorithm.
-
-The outer product of the row and column label vectors shows a
-representation of the checkerboard structure.
+This example demonstrates how to generate a checkerboard dataset and bicluster
+it using the Spectral Biclustering algorithm.
 
+The data is generated, then shuffled and passed to the Spectral Biclustering
+algorithm. The rows and columns of the shuffled matrix are then rearranged to
+plot the biclusters found by the algorithm.
 """
 
 # Author: Kemal Eren <kemal@kemaleren.com>
 # License: BSD 3 clause
 
-import numpy as np
-from matplotlib import pyplot as plt
+# %%
+# Generate sample data
+# --------------------
+# We generate the sample data using the `make_checkerboard` function. Each pixel
+# within `shape=(300, 300)` represents with it's color a value from a uniform
+# distribution. The noise is added from a normal distribution, where the value
+# chosen for `noise` is the standard deviation.
+#
+# As you can see, the data is distributed over 12 cluster cells and is
+# relatively well distinguishable.
 
 from sklearn.datasets import make_checkerboard
-from sklearn.cluster import SpectralBiclustering
-from sklearn.metrics import consensus_score
-
+from matplotlib import pyplot as plt
 
 n_clusters = (4, 3)
 data, rows, columns = make_checkerboard(
-    shape=(300, 300), n_clusters=n_clusters, noise=10, shuffle=False, random_state=0
+    shape=(300, 300), n_clusters=n_clusters, noise=10, shuffle=False, random_state=1
```

Comment:
any reason you're changing the random state here?

Reply:
Yes, I was worried someone could mistake it for leakage. What do you think?

Reply:
nah, I don't think it's an issue. You can also change all random state to 42 :D

Reply:
If we got the answer to all questions, we can't be wrong. ;)